src/logic/Utils.ts

```diff
@@ -307,43 +306,34 @@ export const initColorShiftLoop = (schemes: SchemeIni) => {
   }, 60 * 1000);
 };
 
-export const getColorFromImage = async (image: HTMLImageElement, numColors: number) => {
-  const swatches = await Vibrant.from(image).maxColorCount(numColors).getPalette((err, palette) => {
-    if (err) {
-      console.error(err);
-      return;
-    }
-    return palette;
-  });
-
-  if (swatches.Vibrant) {
-    // remove the # from the hex
-    return swatches.Vibrant.hex.substring(1);
-  }
-
-  return "null";
+export const getColorFromImage = async (image: string) => {
+  const color = await (await Spicetify.colorExtractor(image)).VIBRANT_NON_ALARMING;
+  return color.substring(1);
 };
 
 export const generateColorPalette = async (mainColor: string, numColors: number) => {
   // Generate a palette from https://www.thecolorapi.com/id?hex=0047AB&rgb=0,71,171&hsl=215,100%,34%&cmyk=100,58,0,33&format=html
+  console.log(mainColor);
   const mode = getLocalStorageDataFromKey(LOCALSTORAGE_KEYS.albumArtBasedColorMode);
   // Add a hyphen before any uppercase characters
   const modeStr = mode.replace(/([A-Z])/g, "-$1").toLowerCase();
   //fetch `https://www.thecolorapi.com/scheme?hex=${mainColor}&mode=${modeStr}&count=${numColors}`
   const palette = await  fetch(`https://www.thecolorapi.com/scheme?hex=${mainColor}&mode=${modeStr}&count=${numColors}`)
     .then((response) => response.json());
   // create an array of the hex values for the colors while also removing the #
+  console.log(palette);
   const colorArray = palette.colors.map((color) => color.hex.value.substring(1));
   return colorArray;
 };
 
-async function waitForAlbumArt(): Promise<HTMLImageElement | null> {
+async function waitForAlbumArt(): Promise<string | undefined> {
   // Only return when the album art is loaded
   return new Promise((resolve) => {
     setInterval(() => {
       const albumArt: HTMLImageElement | null = document.querySelector(".main-image-image.cover-art-image");
+      const albumArtSrc: string | undefined = albumArt?.src;
```

Comment:
Why not use internal functions for this? Especially now you'll only need the URL.
```suggestion
      const albumArt = Spicetify.Player.data?.track?.metadata?.image_xlarge_url;
```